compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertLoopsForVectorization.cpp

```diff
@@ -97,8 +105,8 @@ class AMDAIEInsertLoopsForVectorizationPass
   /// this is optinal we need not return failure if the collapsing cannot take
   /// place. Eg: For <2x3x4> since there aren't any unit dimensions, it'd return
   /// failure, hence we can simply return.
-  static void collapseUnitDims(IRRewriter &rewriter,
-                               linalg::GenericOp &genericOp) {
+  void collapseUnitDims(IRRewriter &rewriter, linalg::GenericOp &genericOp) {
```

Comment:
nit: I think the code will be clearer if the check is done at the 2 places the function is called, ie outside the function. i.e. 


```
if (enableCollapsingUnitDims) collapseUnitDims(rewriter, genericOp);
```